src/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -848,7 +848,37 @@ public void Ping_TimedOutSync_Success()
 
         [Fact]
         [OuterLoop]
-        public async Task PingAsync_TimedOutSync_Success()
+        public async Task Ping_TimedOut_EAP_Success()
+        {
+            var sender = new Ping();
+            sender.PingCompleted += (s, e) =>
+            {
+                var tcs = (TaskCompletionSource<PingReply>)e.UserState;
+
+                if (e.Cancelled)
+                {
+                    tcs.TrySetCanceled();
+                }
+                else if (e.Error != null)
+                {
+                    tcs.TrySetException(e.Error);
+                }
+                else
+                {
+                    tcs.TrySetResult(e.Reply);
+                }
+            };
+
+            var tcs = new TaskCompletionSource<PingReply>();
+            sender.SendAsync(TestSettings.UnreachableAddress, tcs);
+
+            PingReply reply = await tcs.Task;
+            Assert.Equal(IPStatus.TimedOut, reply.Status);
+        }
+
+        [Fact]
+        [OuterLoop]
+        public async Task Ping_TimedOut_TAP_Success()
```

Comment:
At first I didn't understand "TAP", but it is [documented](https://docs.microsoft.com/en-us/dotnet/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap) official for "Task-based Async Pattern".  So, I'm good with this. :)